qiskit_optimization/algorithms/qrao/quantum_random_access_optimizer.py

```diff
@@ -0,0 +1,304 @@
+        return self._encoding
+
+    @property
+    def relaxed_fval(self) -> float:
+        """The optimal function value of the relaxed problem."""
+        return self._relaxed_fval
+
+    @property
+    def relaxed_result(
+        self,
+    ) -> MinimumEigensolverResult:
+        """The result obtained from the underlying minimum eigensolver."""
+        return self._relaxed_result
+
+    @property
+    def rounding_result(self) -> RoundingResult:
+        """The rounding result."""
+        return self._rounding_result
+
+
+class QuantumRandomAccessOptimizer(OptimizationAlgorithm):
+    """Quantum Random Access Optimizer class."""
+
+    def __init__(
+        self,
+        min_eigen_solver: MinimumEigensolver,
+        penalty: Optional[float] = None,
+        max_vars_per_qubit: int = 3,
+        rounding_scheme: Optional[RoundingScheme] = None,
+    ):
+        """
+        Args:
+            min_eigen_solver: The minimum eigensolver to use for solving the relaxed problem.
+            penalty: The factor that is used to scale the penalty terms corresponding to linear
+                equality constraints. If ``None`` is provided, the penalty will be automatically
+                determined.
+            rounding_scheme: The rounding scheme.  If ``None`` is provided,
+                ``SemideterministicRounding()`` will be used.
+
+        """
+        self.min_eigen_solver = min_eigen_solver
+        self.penalty = penalty
+        # Use ``QuadraticProgramToQubo`` to convert the problem to a QUBO.
+        self._converters: QuadraticProgramConverter = QuadraticProgramToQubo(penalty=penalty)
+        self._max_vars_per_qubit = max_vars_per_qubit
+        self.rounding_scheme = rounding_scheme
+
+    @property
+    def min_eigen_solver(self) -> MinimumEigensolver:
+        """Return the minimum eigensolver."""
+        return self._min_eigen_solver
+
+    @min_eigen_solver.setter
+    def min_eigen_solver(self, min_eigen_solver: MinimumEigensolver) -> None:
+        """Set the minimum eigensolver."""
+        if not min_eigen_solver.supports_aux_operators():
+            raise TypeError(
+                f"The provided MinimumEigensolver ({type(min_eigen_solver)}) "
+                "does not support auxiliary operators."
+            )
+        self._min_eigen_solver = min_eigen_solver
+
+    @property
+    def max_vars_per_qubit(self) -> int:
+        """Return the maximum number of variables per qubit."""
+        return self._max_vars_per_qubit
+
+    @max_vars_per_qubit.setter
+    def max_vars_per_qubit(self, max_vars_per_qubit: int) -> None:
+        """Set the maximum number of variables per qubit."""
+        self._max_vars_per_qubit = max_vars_per_qubit
+
+    @property
+    def rounding_scheme(self) -> RoundingScheme:
+        """Return the rounding scheme."""
+        return self._rounding_scheme
+
+    @rounding_scheme.setter
+    def rounding_scheme(self, rounding_scheme: RoundingScheme) -> None:
```

Comment:
```suggestion
    def rounding_scheme(self, rounding_scheme: RoundingScheme | None) -> None:
```
As None get passed through from constructor default I think it should show None especially as it especially handles that case. And by noting it here it will allow a user to set it back to the default easily enough.